ProcessMaker/processmaker
# FOUR-10119: Asset Importer

## Issue & Reproduction Steps

As part of our ongoing efforts to improve project asset management, we enhanced the asset import functionality. 

Users will be able to conveniently import assets directly into a project. This enhancement aims to simplify the process by automating asset type detection and importing specific assets to the Project.

## Solution
- Implemented an import functionality within each Project.
- Identify what type of asset is been imported and attach it to the Project
- Created unitTest to ensure the proper functionality of the feature.

## How to Test
- Please checkout to task/FOUR-10119 on the following.
-- Core
-- Package Projects
-- Decision Engine
- Navigate to Designer and create a Project
- Click on the Import button 
- You can now Import the following type of assets:
-- Projects
-- Screens
-- Decision Tables

Please ensure that the Assets have been imported to the Project and make sure we can utilize the Custom importer as well.

## Related Tickets & Packages
- Ticket : [FOUR-10119](https://processmaker.atlassian.net/browse/FOUR-10119)
- [Core](https://github.com/ProcessMaker/processmaker/pull/5424)
- [Package Projects](https://github.com/ProcessMaker/package-projects/pull/37) 
- [Decision Engine](https://github.com/ProcessMaker/package-decision-engine/pull/71)

## CI 
ci:next
ci:package-projects:task/FOUR-10119
ci:package-decision-engine:task/FOUR-10119


## Code Review Checklist


[FOUR-10119]: https://processmaker.atlassian.net/browse/FOUR-10119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
